cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -118,9 +119,6 @@ type ClusterConfiguration struct {
 
 	// FeatureGates enabled by the user.
 	FeatureGates map[string]bool
-
-	// The cluster name
-	ClusterName string
```

Comment:
Missing many api-machinery context, so out of curiosity, would keeping `ClusterName` in the internal type make sense or be possible? (and not having `ObjectMeta` in the internal type)

Reply:
That's a good idea. We can do this now that we use custom ObjectMeta, containing only the name field.
However, `metav1.ObjectMeta` has a `ClusterName` field with a slightly more different context and this can cause problems with we choose to go with an expanded ObjectMeta or even the `metav1` one. That's why I originally went this way.

Reply:
one question we might ask our selfs is do we really need ObjectMeta in the internal type, i.e. would we need values from it after external is converted to internal?
ObjectMeta only conforms Kustomize for external types.

but i don't mind either way at this point.

Reply:
> In the ClusterConfiguration, `metadata.name` replaces the `clusterName` field.
The latter is therefore removed. Hence, the following v1beta2
ClusterConfiguration is upgraded to v1beta3 as such:

this might trip some users that don't use "migrate", but they should.